File: examples/go/order-specification/order/order.go
package order

import "time"

type Order struct {
	id           string
	paid         bool
	refunded     bool
	placedAt     time.Time
	refundCutoff time.Time
}

func New(id string, paid bool, refunded bool, placedAt time.Time, refundCutoff time.Time) Order {
	return Order{
		id:           id,
		paid:         paid,
		refunded:     refunded,
		placedAt:     placedAt,
		refundCutoff: refundCutoff,
	}
}

func (o Order) ID() string {
	return o.id
}

func (o Order) IsPaid() bool {
	return o.paid
}

func (o Order) IsRefunded() bool {
	return o.refunded
}

func (o Order) PlacedAt() time.Time {
	return o.placedAt
}

func (o Order) RefundCutoff() time.Time {
	return o.refundCutoff
}

File: examples/typescript/order-specification/order/order.ts
export class Order {
  private constructor(
    private readonly idValue: string,
    private readonly paidValue: boolean,
    private readonly refundedValue: boolean,
    private readonly placedAtValue: Date,
    private readonly refundCutoffValue: Date
  ) {}

  static create(
    id: string,
    paid: boolean,
    refunded: boolean,
    placedAt: Date,
    refundCutoff: Date
  ): Order {
    return new Order(id, paid, refunded, placedAt, refundCutoff);
  }

  id(): string {
    return this.idValue;
  }

  isPaid(): boolean {
    return this.paidValue;
  }

  isRefunded(): boolean {
    return this.refundedValue;
  }

  placedAt(): Date {
    return this.placedAtValue;
  }

  refundCutoff(): Date {
    return this.refundCutoffValue;
  }
}
